superset-frontend/src/explore/components/DatasourcePanel/index.tsx

```diff
@@ -190,6 +217,35 @@ export default function DataSourcePanel({
     [_columns],
   );
 
+  const placeholderSlDataset = {
+    sl_table: [],
+    query: [],
+    saved_query: [],
+  };
+
+  // eslint-disable-next-line no-param-reassign
+  datasource.sl_dataset = placeholderSlDataset;
```

Comment:
I think the value that you're looking for is `datasource.type` which will come from the api.

Reply:
This was the hard coded placeholder that I discussed with you the other day in Slack, but I ended up removing it in [`this commit`](https://github.com/apache/superset/pull/19855/commits/fd65bfa6f6241c466fa4eac90efafcc87f5c95cd) because the example still works. Should I keep it as is or hard code a placeholder again?